File: blazerod/render/src/main/java/top/fifthlight/armorstand/extension/CommandEncoderExt.java
package top.fifthlight.armorstand.extension;

import com.mojang.blaze3d.buffers.GpuBufferSlice;

public interface CommandEncoderExt {
    enum ClearType {
        ZERO_FILLED,
        BYTE_ONE_FILLED,
        FLOAT_ONE_FILLED,
    }

    void armorStand$clearBuffer(GpuBufferSlice slice, ClearType clearType);

    void armorStand$copyBuffer(GpuBufferSlice target, GpuBufferSlice source);
}

File: blazerod/render/src/main/kotlin/top/fifthlight/armorstand/extension/CommandEncoderExt.kt
package top.fifthlight.armorstand.extension

import com.mojang.blaze3d.buffers.GpuBufferSlice
import com.mojang.blaze3d.systems.CommandEncoder

fun CommandEncoder.clearBuffer(slice: GpuBufferSlice, clearType: CommandEncoderExt.ClearType) =
    (this as CommandEncoderExt).`armorStand$clearBuffer`(slice, clearType)

fun CommandEncoder.copyBuffer(target: GpuBufferSlice, source: GpuBufferSlice) =
    (this as CommandEncoderExt).`armorStand$copyBuffer`(target, source)
